todoman/model.py

```diff
@@ -87,263 +92,245 @@ def __init__(self, todo=None, filename=None, mtime=None, new=False):
         be populated with default values.
         """
 
-        if todo:
-            self.todo = todo
-        else:
-            self.todo = icalendar.Todo()
+        self._attributes = {}
+
+        now = datetime.now(LOCAL_TIMEZONE)
+        self.uid = '{}@{}'.format(uuid4().hex, socket.gethostname())
 
         if new:
-            now = datetime.now(self._localtimezone)
-            uid = uuid4().hex + socket.gethostname()
-            self.todo.add('uid', uid)
-            self.todo.add('percent-complete', 0)
-            self.todo.add('priority', 0)
-            self.todo.add('created', now)
+            self.created_at = now
 
-        if self.todo.get('dtstamp', None) is None:
-            self.todo.add('dtstamp', datetime.utcnow())
+        if not self.dtstamp:
+            self.dtstamp = now
+
+        self.filename = filename or "{}.ics".format(self.uid)
 
-        self.filename = filename or "{}.ics".format(self.todo.get('uid'))
         if os.path.basename(self.filename) != self.filename:
-            raise ValueError('Must not be an absolute path: {}'
-                             .format(self.filename))
+            raise ValueError(
+                'Must not be an absolute path: {}' .format(self.filename)
+            )
         self.mtime = mtime or datetime.now()
 
-    def _set_field(self, name, value):
-        if name in self.todo:
-            del(self.todo[name])
-        # We want to save things like [] or 0, but not null, or nullstrings
-        if value is not None and value is not '':
-            logger.debug("Setting field %s to %s.", name, value)
-            self.todo.add(name, value)
-
-    @property
-    def status(self):
-        return self.todo.get('status', 'NEEDS-ACTION')
-
-    @status.setter
-    def status(self, value):
-        self._set_field('status', value)
+    DEFAULT_VALUES = {
+        'categories': [],
+        'completed_at': None,
+        'created_at': None,
+        'description': '',
+        'dtstamp': None,
+        'start': None,
+        'due': None,
+        'location': '',
+        'percent_complete': 0,
+        'priority': 0,
+        'sequence': 0,
+        'status': 'NEEDS-ACTION',
+        'summary': '',
+        'uid': None,
+        'id': None,  # Not the right place
+    }
+
+    KNOWN_FIELDS = DEFAULT_VALUES.keys()
+
+    def __getattr__(self, name):
+        if name in Todo.KNOWN_FIELDS:
+            if name in self._attributes:
+                return self._attributes[name]
+            else:
+                return Todo.DEFAULT_VALUES[name]
+        raise AttributeError("'{}' has no attribute '{}'".format(self, name))
+
+    def __setattr__(self, name, value):
+        if name in Todo.KNOWN_FIELDS:
```

Comment:
Trying to do this actually unveiled how fragile all the dealing with defaults is (eg: priority 0 vs None, etc).

I'll put some work into this, since I don't like this fragility.